numpy/distutils/fcompiler/environment.py

```diff
@@ -55,9 +55,11 @@ def _get_var(self, name, conf_desc):
         if envvar is not None:
             envvar_contents = os.environ.get(envvar)
             if envvar_contents is not None:
-                if var and append:
+                if convert:
+                    envvar_contents = convert(envvar_contents)
+                if var and append: # in case var is None?
```

Comment:
I think more for cases when `var == []`, in which case emitting a warning about append vs overwrite is a waste of time.

Reply:
That's a valid point, but from the correctness of the code, the relevanc is whehter None a valid return value of the hook function or not (a few lines above). Do you know?

Reply:
I don't know the answer to that. I know that var is sometimes `[]`. Your comment loosely implies that `[]` can never happen, so I think does more harm than good. Dropping the comment you added would be fine by me.

Reply:
Indeed.